### PR TITLE
Ol clustering

### PR DESCRIPTION
**Pull request by RobertOrthofer**
This PR includes the initial version of the OL clustering in the center panel.

Additional functionality, like routing via map interactions or additional styling of single features are not included in this PR.
Also, component cleanups in lifecycle-hooks are not entirely implemented due to the complex data- and application structure.

On a side note, the node version for this project has been set to v14.x, due to compatibility issues with v16.x

Closes https://github.com/eurodatacube/eodash/issues/1193

### CODE REVIEW

**Comment by silvester-pari** [comment]
Thanks @RobertOrthofer, looks great so far! Tested it and it works well, and it's a great start for the migration.
I changed the PR base to a new branch `leaflet-openlayers-migration` where we can prepare everything regarding the migration and merge everything to `staging` at once when we're done. Also, this new branch in the eodash repository gives us a deployment for the feature branch to make testing easier.

Do you prefer to merge this to `leaflet-openlayers-migration` already or to continue your work (styling of individual features, click events etc.) in this branch?

Btw. I saw just a minor linter error in `olMapHelpers.js` ("`110:1  error  This line has a length of 104. Maximum allowed is 100`"), but it's probably best to do a linting session once all the features are there.

**Comment by RobertOrthofer** [comment]
Thank you for the feedback!
Sorry for the branching mistake. 

I can work in this branch without merging for now, map interactions and additional styling would to this first topic of clustering.
About the linting: I was already wondering if you got some global `eslint`-configuration, or some general coding guidelines i have to follow? Also concerning naming conventions and general document structure.

**Comment by silvester-pari** [comment]
No worries, no mistakes were made :-)

Regarding linting, we have the https://github.com/eurodatacube/eodash/blob/staging/app/.eslintrc.js file with some linting rules, although they do not (yet) follow our global company rules for vue.js linting (this project is simply too old for that). In the future, we'll want to upgrade to our centralized eslint config, but this is something that is not of great importance now.

Other than that, your code looks really good! Only thing I was wondering is that the `package-lock.json` now uses a `lockfileVersion 2` (from npm >7), although we use node 14 (indicating npm <7), so that is quite a substatial diff. Either it can be reduced in the PR by using an older npm version that creates a `lockfileVersion 1`, or we all move to a newer npm version with `lockfileVersion 2`.

Thumbs up from my side to continue working in this branch/PR!

**Comment by ahocevar** [comment]
@RobertOrthofer When you open your VSCode editor from the `app/` folder, it should pick up the `.eslintrc` file and show any problems in the "Problems" section.

@silvester-pari It could make sense to set up a GitHub action to run linter and tests for pull requests against any branch. Please let us know if you want us to set that up.

**Comment by silvester-pari** [comment]
@ahocevar thanks, great idea! We had GitHub actions for feature branches so far, but not for pull requests from forks - I added a lint & test action that should trigger on every pull request.

All workflows in the project currently run on node 12, so if we decide together to go fore e.g. node 14 we should update the workflows accordingly. I remember some incompatible npm packages, so we went for node 12 in the beginning, not sure if that still applies. But it might be good to discuss this in a separate issue if you suggest to upgrade the node version.

We don't really have tests in this project for now (except for the custom dashboard feature), which is probably a huge no-go 🙈
It's on our todo-list...

**Comment by ahocevar** [comment]
Given the high activity in this repository, I think it would be good to merge early and often. That said, I would suggest that @RobertOrthofer sets the node version back to v12 for now (if that works without any problems), and this pull request be merged into `leaflet-openlayers-migration`. And `leaflet-openlayers-migration` should be kept up to date with `staging`. Otherwise we risk that `leaflet-openlayers-migration` deviates too much from `staging` to be successfully merged when the migration is complete.

**Comment by ahocevar** [comment]
@silvester-pari @RobertOrthofer I just pushed two commits to go back to node v12 / lockfile v1 for now, and to remove the lint you mentioned.

**Comment by ahocevar** [comment]
@silvester-pari I really think it is time to rebase `leaflet-openlayers-migration` to staging, so @RobertOrthofer can rebase this pull request to `leaflet-openlayers-migration`. I fear we'll be getting unresolvable merge conflicts when we don't rebase often.

**Comment by silvester-pari** [comment]
Sorry for my late reply, yes sure that's a good idea! But wouldn't it be enough to just merge staging into the branch periodically?

**Comment by ahocevar** [comment]
@silvester-pari 
> But wouldn't it be enough to just merge staging into the branch periodically?

Yes, that's what we'd ask you to do. We'll take care of rebasing our feature branches, which will also be necessary if there are changes in `leaflet-openlayers-migration` that affect parts of our feature branches. Shouldn't happen too often though, unless we have long-living feature branches.

**Comment by silvester-pari** [comment]
Got it. `leaflet-openlayers-migration` is now up to date. Will try to update weekly from now on!

**Comment by RobertOrthofer** [comment]
@silvester-pari looks like i need a review from a maintainer as a first-time contributer. All of this is still kind of WIP, but i would use this as a base for the globe i have in another branch.

**Comment by ahocevar** [comment]
@RobertOrthofer has now added the final touch-ups, and I have finished my review. @silvester-pari, this is now ready for you to review.

File: app/src/components/map/olMapHelpers.js
Comment (silvester-pari) [inline app/src/components/map/olMapHelpers.js:49]: Here it would be good to use the default colors as defined in https://github.com/w3geo/eodash/blob/ol-clustering/app/src/components/Map.vue#L233-L240

**Review by silvester-pari — CHANGES_REQUESTED** [review]
This looks like a good start, thanks! If I understand correctly, you'd want to merge this into the `leaflet-openlayers-migration` but keep the branch open for further work (e.g. click action connecting to the POI data)?

**Comment by silvester-pari** [comment]
Sorry for the failing Lint and Test check, the way it is currently set up it seems to only work for branches inside the repository...

I would say regarding this and the merge in general we'll wait for the outcome of tomorrow's meeting, in the meantime I would be happy to hear your thoughts on my comment from above regarding the material design icons!

File: app/src/components/map/olMapHelpers.js
Comment (RobertOrthofer) [reply app/src/components/map/olMapHelpers.js]: of course, i fixed this with [9c13827](https://github.com/eurodatacube/eodash/pull/1204/commits/9c138278c614e4874a1c794388f4491fb2178f3e).

**Comment by silvester-pari** [comment]
Thanks for the changes! Looking good now. I will do the merge into `leaflet-openlayers-migration` as agreed.

**Review by silvester-pari — APPROVED** [review]

**Comment by silvester-pari** [comment]
There seems only to be a merge conflict in package-lock.json since we updated the marked package, otherwise looking good to go!

**Comment by ahocevar** [comment]
Merge conflicts resolved.

**Pull request merged**
